sdk/search/Azure.Search.Documents/apis.txt

```diff
@@ -31,34 +31,34 @@ namespace Azure.Search.Documents
 
         Response<AutocompleteResults> Autocomplete(string, string, AutocompleteOptions, CancellationToken) { }
         Task<Response<AutocompleteResults>> AutocompleteAsync(string, string, AutocompleteOptions, CancellationToken) { }
-        
+
         Response<IndexDocumentsResult> IndexDocuments(IndexDocumentsBatch<SearchDocument>, IndexDocumentsOptions, CancellationToken) { }
         Task<Response<IndexDocumentsResult>> IndexDocumentsAsync(IndexDocumentsBatch<SearchDocument>, IndexDocumentsOptions, CancellationToken) { }
         Response<IndexDocumentsResult> IndexDocuments<T>(IndexDocumentsBatch<T>, IndexDocumentsOptions, CancellationToken) { }
         Task<Response<IndexDocumentsResult>> IndexDocumentsAsync<T>(IndexDocumentsBatch<T>, IndexDocumentsOptions, CancellationToken) { }
     }
 
-    public class FieldBuilder { }
-    // TODO: FieldBuilder's attributes like [Key], [Searchable], etc.
+    public class SearchIndexBuilder { }
+    // TODO: SearchIndexBuilder's attributes like [Key], [Searchable], etc.
 
     public class SearchServiceClient
     {
+        SearchClient GetSearchClient(string) { }
```

Comment:
I think these should go into the Indexes namespace - all except `SearchClient` and `SearchServiceClient`. The idea was to hide the majority of indexing, and they are included when you use one of these members.

Reply:
The client types themselves are already in the `.Indexes` namespace - it may not have been super obvious.

We could consider moving all these `Get*Client` accessors there as well via extension methods.  There aren't so many APIs on `SearchServiceClient` though that I think it's a problem here - and it makes the workflow of "start with a C# object, add some attributes, and easily create an index" a tiny bit easier.

Reply:
What would be the advantage of using extension methods, especially when other languages don't have the same luxury? They certainly reduce the number of methods. More recent VSes will add those extension methods anyway without using the namespace since they are already in a referenced assembly.

Reply:
We did that in Storage so you weren't exposed to `.Specialized` blob types from the main clients unless you pulled in the `.Specialized` namespace.  I'm less concerned about it here because we don't have "regular indexing" and "advanced indexing" - all our indexing is equally complicated.